huaweicloud/services/acceptance/obs/resource_huaweicloud_obs_bucket_replication_test.go

```diff
@@ -0,0 +1,167 @@
+		Steps: []resource.TestStep{
+			{
+				Config: testAccObsBucketReplication_basic(agencyName, bucketName, acceptance.HW_OBS_DESTINATION_BUCKET),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(rName, "bucket", bucketName),
+					resource.TestCheckResourceAttr(rName, "agency", agencyName),
+					resource.TestCheckResourceAttr(rName, "destination_bucket", acceptance.HW_OBS_DESTINATION_BUCKET),
+					resource.TestCheckResourceAttr(rName, "rule.#", "1"),
+					resource.TestCheckResourceAttr(rName, "rule.0.enabled", "true"),
+					resource.TestCheckResourceAttr(rName, "rule.0.prefix", "abc"),
+					resource.TestCheckResourceAttrSet(rName, "rule.0.id"),
+				),
+			},
+			{
+				Config: testAccObsBucketReplication_update_1(agencyName, bucketName, acceptance.HW_OBS_DESTINATION_BUCKET),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(rName, "rule.#", "2"),
+					resource.TestCheckResourceAttr(rName, "rule.0.enabled", "true"),
+					resource.TestCheckResourceAttr(rName, "rule.0.prefix", "imgs/"),
+					resource.TestCheckResourceAttr(rName, "rule.1.enabled", "false"),
+					resource.TestCheckResourceAttr(rName, "rule.1.prefix", "terraform"),
+					resource.TestCheckResourceAttr(rName, "rule.1.storage_class", "COLD"),
+					resource.TestCheckResourceAttr(rName, "rule.1.history_enabled", "true"),
+					resource.TestCheckResourceAttrSet(rName, "rule.0.id"),
+					resource.TestCheckResourceAttrSet(rName, "rule.1.id"),
+				),
+			},
+			{
+				Config: testAccObsBucketReplication_update_2(agencyName, bucketName, acceptance.HW_OBS_DESTINATION_BUCKET),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(rName, "rule.#", "1"),
+					resource.TestCheckResourceAttr(rName, "rule.0.prefix", ""),
+					resource.TestCheckResourceAttr(rName, "rule.0.enabled", "true"),
+					resource.TestCheckResourceAttrSet(rName, "rule.0.id"),
+				),
+			},
+			{
+				ResourceName:      rName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccObsBucketReplication_base(agencyName, bucketName string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_identity_agency" "agency" {
+  name                   = "%s"
+  description            = "This is a iam agency for obs bucket replication"
+  delegated_service_name = "op_svc_obs"
+
+  domain_roles = [
+    "OBS Administrator",
+  ]
+}
+
+resource "huaweicloud_obs_bucket" "source" {
+  bucket        = "%s"
+  storage_class = "STANDARD"
+  acl           = "private"
+}
+`, agencyName, bucketName)
+}
+
+func testAccObsBucketReplication_basic(agencyName, bucketName, destinationName string) string {
+	return fmt.Sprintf(`
+%s
+
+resource "huaweicloud_obs_bucket_replication" "replica" {
+  bucket             = huaweicloud_obs_bucket.source.bucket
+  destination_bucket = "%s"
```

Comment:
can we create another bucket?

Reply:
I will create a bucket with a target region.